### ISSUE TITLE
XtextDocumentModifyTest / XtextGrammarQuickfixTest fails on windows

### ISSUE THREAD

**Issue by cdietrich** [eclipse/xtext-eclipse#1539]
XtextDocumentModifyTest / XtextGrammarQuickfixTest fails on windows

i assume that the (test) api is line ending dependent somehow

XtextGrammarQuickfixTest actually seems to be flaky
maybe https://github.com/eclipse/xtext-eclipse/issues/1022 fix is not properly working
as also the label is same?

```

Failures:
  XtextGrammarQuickfixTest.test_convert_terminal_fragment_to_terminal_rule:85->AbstractQuickfixTest.testQuickfixesOn:176->AbstractQuickfixTest.quickfixesAreOffered:217->AbstractQuickfixTest.assertIssueResolutionResult:259->Assert.assertEquals:146->Assert.assertEquals:117 expected:<...example/mydsl/MyDsl"[

]odel hidden():
        a = ID;

termina...> but was:<...example/mydsl/MyDsl"[

Model hidden():]
        a = ID;

termina...>
  XtextGrammarQuickfixTest.test_fix_invalid_hidden_token:171->AbstractQuickfixTest.testQuickfixesOn:176->AbstractQuickfixTest.quickfixesAreOffered:217->AbstractQuickfixTest.assertIssueResolutionResult:259->Assert.assertEquals:146->Assert.assertEquals:117 expected:<...example/mydsl/MyDsl"[

]odel hidden():
        a = ID;

Another...> but was:<...example/mydsl/MyDsl"[

Model hidden():]
        a = ID;

Another...>

Tests run: 265, Failures: 2, Errors: 0, Skipped: 1
````

**Comment by cdietrich** [comment]
[org.eclipse.xtext.ui.tests.editor.model.edit.XtextDocumentModifyTest.txt](https://github.com/eclipse/xtext-eclipse/files/5244936/org.eclipse.xtext.ui.tests.editor.model.edit.XtextDocumentModifyTest.txt)
[org.eclipse.xtext.xtext.ui.editor.quickfix.XtextGrammarQuickfixTest.txt](https://github.com/eclipse/xtext-eclipse/files/5244945/org.eclipse.xtext.xtext.ui.editor.quickfix.XtextGrammarQuickfixTest.txt)

**Comment by cdietrich** [comment]
XtextDocumentModifyTest is fixed

**Comment by cdietrich** [comment]
there is the problem that org.eclipse.xtext.formatting.impl.AbstractDeclarativeFormatter.getConfig()
uses a shared configuration from a singleton store
thus the first guess for "\n" as line separator will be used for everything

**Comment by cdietrich** [comment]
fixed in 2.24
